optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,53 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.4.0")
+class BestValueStagnationEvaluator(BaseImprovementEvaluator):
+    """
+    Evaluates the stagnation period of the best value in an optimization process.
+
+    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
+    and is designed toevaluate the remaining trials before reaching this maximum period
+    of allowed stagnation.
```

Comment:
@g-votte What do you think of this class name?
If you have time, could you join the review, please?